pkg/tsdb/interval/interval.go

```diff
@@ -53,16 +54,35 @@ func (i *Interval) Milliseconds() int64 {
 	return i.Value.Nanoseconds() / int64(time.Millisecond)
 }
 
-func (ic *intervalCalculator) Calculate(timerange plugins.DataTimeRange, minInterval time.Duration) Interval {
+func (ic *intervalCalculator) Calculate(timerange plugins.DataTimeRange, interval time.Duration, intervalMode string) Interval {
 	to := timerange.MustGetTo().UnixNano()
 	from := timerange.MustGetFrom().UnixNano()
-	interval := time.Duration((to - from) / defaultRes)
+	calculatedInterval := time.Duration((to - from) / defaultRes)
+
+	switch intervalMode {
+	case "min":
+		if calculatedInterval < interval {
+			return Interval{Text: FormatDuration(interval), Value: interval}
+		}
+	case "max":
+		if calculatedInterval > interval {
+			return Interval{Text: FormatDuration(interval), Value: interval}
+		}
+	case "exact":
+		return Interval{Text: FormatDuration(interval), Value: interval}
```

Comment:
i recommend putting in a `default` case that explodes 💥 , to catch cases when people make a typo in that string, maybe even `panic()` there, like here for example:
https://github.com/grafana/grafana/blob/main/pkg/tsdb/postgres/tlsmanager.go#L92-L101

Reply:
hmm.. or if not `panic()` then at least logging something to the error-log (i'm not sure what is the policy on `panic()`, maybe check with backend-team?)

Reply:
I wouldn't panic, I would log the error or return it